Sources/SwiftKuery/ConnectionPool.swift

```diff
@@ -23,8 +23,26 @@ import Dispatch
 
 // MARK: ConnectionPool
 
-/// A connection pool implementation.
-/// The pool is FIFO.
+/**
+ This class implements a First in First out connection pool. The pool maintains a cache of `ConnectionPoolConnection` instances, which can be reused for future requests, to enhance performance.
+ ### Usage Example: ###
+ In this example, a closure to generate and release `Connection` instances are defined.
+ A connection pool is then initialized with defined options, the `connectionGenerator` and the `connectionReleaser`.
+ A single `Connection` is then retrieved from the pool.
+ ```swift
+ let options = ConnectionPoolOptions(initialCapacity: 2, maxCapacity: Int = 5, timeout: Int = 1000)
```

Comment:
Need to remove `Int =` from the code as this doesn't compile, it needs to be `maxCapacity: 5, timeout: 1000`